relay-general/src/pii/minidumps.rs

```diff
@@ -130,23 +253,157 @@ impl PiiAttachmentsProcessor<'_> {
         data: &mut [u8],
     ) -> Result<bool, ScrubMinidumpError> {
         let file_state = self.state(filename, ValueType::Minidump);
-        let streams = MinidumpData::parse(data)?.streams()?;
+        let items = MinidumpData::parse(data)?.items()?;
         let mut changed = false;
 
-        for StreamDescriptor(stream_type, value_type, range) in streams {
-            let slice = data
-                .get_mut(range)
-                .ok_or(ScrubMinidumpError::InvalidAddress(stream_type))?;
-
+        for item in items {
             // IMPORTANT: Minidump sections are always classified as Pii:Maybe. This avoids to
             // accidentally scrub stack memory with highly generic selectors. TODO: Update the PII
             // system with a better approach.
             let attrs = Cow::Owned(FieldAttrs::new().pii(Pii::Maybe));
 
-            let state = file_state.enter_static("", Some(attrs), Some(value_type));
-            changed |= self.scrub_bytes(slice, &state);
+            match item {
+                MinidumpItem::StackMemory(range) => {
+                    let slice = data
+                        .get_mut(range)
+                        .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let state =
+                        file_state.enter_static("", Some(attrs), Some(ValueType::StackMemory));
+                    changed |= self.scrub_bytes(slice, &state, ScrubEncodings::All);
+                }
+                MinidumpItem::NonStackMemory(range) => {
+                    let slice = data
+                        .get_mut(range)
+                        .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let state =
+                        file_state.enter_static("", Some(attrs), Some(ValueType::HeapMemory));
+                    changed |= self.scrub_bytes(slice, &state, ScrubEncodings::All);
+                }
+                MinidumpItem::LinuxEnviron(range) | MinidumpItem::LinuxCmdLine(range) => {
+                    let slice = data
+                        .get_mut(range)
+                        .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let state = file_state.enter_static("", Some(attrs), Some(ValueType::Binary));
+                    changed |= self.scrub_bytes(slice, &state, ScrubEncodings::All);
+                }
+                MinidumpItem::CodeModuleName(range) => {
+                    let slice = data
+                        .get_mut(range)
+                        .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    // Mirrors decisions made on NativeImagePath type
+                    let state =
+                        file_state.enter_static("code_file", Some(attrs), Some(ValueType::String));
+                    let wstr = WStr::from_utf16le_mut(slice)?; // TODO: Consider making this lossy?
+                    changed |= self.scrub_utf16_filepath(wstr, &state);
```

Comment:
I think for these you should push a path onto the state that emulates something like `$debug_meta.images[].code_name`.

Reply:
We can also add this at a later stage, this is still G2G